samples/npmx/charger_and_events/src/main.c

```diff
@@ -6,443 +6,49 @@
-
-	ret = npm1300_stop_boot_timer(&pm);
-
-	if (ret) {
-		printk("Boot timer stopped.\r\n");
-	} else {
-		printk("Boot timer stopping failed.\r\n");
-	}
-
-	ret = npm1300_connection_check(&pm);
-
-	if (ret) {
-		printk("Connection check ok.\r\n");
 	} else {
-		printk("Connection check failed.\r\n");
+		printk("NPMX device ok.\n");
 	}
+	
+	/* Temporary tests: get pointer to npmx device */
+	npmx_instance_t * npmx_instance = &((struct npmx_data *)npmx_dev->data)->npmx_instance;
 
-	npm1300_gpio_mode_set(&pm, NPM1300_GPIO_INSTANCE_0, NPM1300_GPIO_MODE_OUTPUT_IRQ);
-
-	npm1300_gpio_mode_set(&pm, NPM1300_GPIO_INSTANCE_1, NPM1300_GPIO_MODE_OUTPUT_PLW);
-
-	npm1300_pof_enable(&pm, NTC1300_POF_POLARITY_HI, NTC1300_POF_THRESHOLD_3V3);
-
-	/* Disable charger before changing charge current */
-	npm1300_charger_module_disable(&pm, NPM1300_CHARGER_MODULE_CHARGER_MASK);
-
-	npm1300_charger_charging_current_set(&pm, 200);
-
-	npm1300_charger_termination_volatge_normal_set(&pm, NPM1300_CHARGER_VOLTAGE_4V10);
-
-	/* Enable charger for events handling */
-	npm1300_charger_module_enable(&pm,
-				      NPM1300_CHARGER_MODULE_CHARGER_MASK
-				      | NPM1300_CHARGER_MODULE_RECHARGE_MASK
-				      | NPM1300_CHARGER_MODULE_NTC_LIMITS_MASK);
-
-	npm1300_event_interrupt_enable(&pm,
-				       NPM1300_EVENT_GROUP_VBUSIN_VOLTAGE,
-				       NPM1300_EVENT_GROUP_VBUSIN_DETECTED_MASK
-				       | NPM1300_EVENT_GROUP_VBUSIN_REMOVED_MASK);
-
-	npm1300_event_interrupt_enable(&pm,
-				       NPM1300_EVENT_GROUP_BAT_CHAR_BAT,
-				       NPM1300_EVENT_GROUP_BATTERY_DETECTED_MASK
-				       | NPM1300_EVENT_GROUP_BATTERY_REMOVED_MASK);
-
-	/* Enable all charging status events */
-	npm1300_event_interrupt_enable(&pm,
-				       NPM1300_EVENT_GROUP_BAT_CHAR_STATUS,
-				       NPM1300_EVENT_GROUP_CHARGER_SUPPLEMENT_MASK
-				       | NPM1300_EVENT_GROUP_CHARGER_TRICKLE_MASK
-				       | NPM1300_EVENT_GROUP_CHARGER_CC_MASK
-				       | NPM1300_EVENT_GROUP_CHARGER_CV_MASK
-				       | NPM1300_EVENT_GROUP_CHARGER_COMPLETED_MASK
-				       | NPM1300_EVENT_GROUP_CHARGER_ERROR_MASK);
-
-	/* Enable ADC measurements ready interrupts */
-	npm1300_event_interrupt_enable(&pm,
-				       NPM1300_EVENT_GROUP_ADC,
-				       NPM1300_EVENT_GROUP_ADC_BAT_READY_MASK);
-
-	/* Enable VBAT auto measurement every 1 second */
-	npm1300_adc_vbat_auto_meas_enable(&pm);
+	/* Temporary tests: register callback for vbus events */
+	npmx_driver_register_cb(vbusin_callback, NPMX_CALLBACK_TYPE_EVENT_VBUSIN_VOLTAGE);
 
-	npm1300_adc_ntc_set(&pm, NPM1300_BATTERY_NTC_TYPE_10_K);
+	/* Temporary tests: use GPIO 0 as interrupt output */
+	npmx_gpio_mode_set(npmx_instance, NPMX_GPIO_INSTANCE_0, NPMX_GPIO_MODE_OUTPUT_IRQ);
 
-	npm1300_check_errors(&pm);
+	/* Temporary tests: enable interrupt from vbusin events */
+	npmx_driver_event_interrupt_enable(NPMX_EVENT_GROUP_VBUSIN_VOLTAGE,
+				       NPMX_EVENT_GROUP_VBUSIN_DETECTED_MASK
+				       | NPMX_EVENT_GROUP_VBUSIN_REMOVED_MASK);
 
```

Comment:
I think or sign `|` should not be at the beginning of the line, but at the end of the previous one instead. Also, arguments should be aligned.